File: demos/android/demos/src/main/java/com/antgroup/antv/f2/samples/charts/PieChart.java
package com.antgroup.antv.f2.samples.charts;

import com.antgroup.antv.f2.F2CanvasView;
import com.antgroup.antv.f2.F2Chart;
import com.antgroup.antv.f2.F2Config;
import com.antgroup.antv.f2.samples.Utils;

/**
 * 饼图
 *
 * @author qingyuan.yl
 * @date 2020-09-27
 */
public class PieChart implements F2CanvasView.Adapter {
    private F2Chart mChart;

    @Override
    public void onCanvasDraw(F2CanvasView canvasView) {
        if (mChart == null) {
            mChart = F2Chart.create(canvasView.getContext(), "PieChart", canvasView.getWidth(), canvasView.getHeight());
        }

        mChart.setCanvas(canvasView);
        mChart.padding(10, 20, 0, 0);
        mChart.source(Utils.loadAssetFile(canvasView.getContext(), "mockData_basePie.json"));
        mChart.setAxis("percent", new F2Chart.AxisConfigBuilder()
                .lineHidden()
                .labelHidden()
                .gridHidden());

        mChart.setAxis("a", new F2Chart.AxisConfigBuilder().labelHidden().lineHidden().gridHidden());
        mChart.setCoord(new F2Chart.CoordConfigBuilder().type("polar").transposed(true));
        mChart.interval().position("a*percent").fixedSize(1).style(new F2Config.Builder().setOption("lineWidth", 3).build()).color("name").adjust("stack");
        mChart.render();
}

    @Override
    public void onDestroy() {
        if (mChart != null) {
            mChart.destroy();
        }
    }
}

File: demos/android/demos-kotlin/src/main/java/com/antgroup/antv/f2/samples/charts/PieChart.kt
package com.antgroup.antv.f2.samples.charts

import com.antgroup.antv.f2.F2CanvasView
import com.antgroup.antv.f2.F2Chart
import com.antgroup.antv.f2.F2Chart.AxisConfigBuilder
import com.antgroup.antv.f2.F2Chart.CoordConfigBuilder
import com.antgroup.antv.f2.F2Config
import com.antgroup.antv.f2.samples.Utils.loadAssetFile

/**
 * 饼图
 *
 * @author qingyuan.yl
 * @date 2020-09-27
 */
class PieChart : F2CanvasView.Adapter {
    private var mChart: F2Chart? = null
    override fun onCanvasDraw(canvasView: F2CanvasView) {
        if (mChart == null) {
            mChart = F2Chart.create(
                canvasView.context,
                "PieChart",
                canvasView.width.toDouble(),
                canvasView.height.toDouble()
            )
        }
        mChart!!.setCanvas(canvasView)
        mChart!!.padding(10.0, 20.0, 0.0, 0.0)
        mChart!!.source(loadAssetFile(canvasView.context, "mockData_basePie.json"))
        mChart!!.setAxis(
            "percent", AxisConfigBuilder()
                .lineHidden()
                .labelHidden()
                .gridHidden()
        )
        mChart!!.setAxis("a", AxisConfigBuilder().labelHidden().lineHidden().gridHidden())
        mChart!!.setCoord(CoordConfigBuilder().type("polar").transposed(true))
        mChart!!.interval().position("a*percent").fixedSize(1f)
            .style(F2Config.Builder<Any?>().setOption("lineWidth", 3).build()).color("name")
            .adjust("stack")
        mChart!!.render()
    }

    override fun onDestroy() {
        if (mChart != null) {
            mChart!!.destroy()
        }
    }
}